charms/worker/k8s/src/snap.py

```diff
@@ -279,10 +279,9 @@ def management(charm: K8sCharmProtocol) -> None:
         if isinstance(args, SnapFileArgument) and which.revision != "x1":
             snap_lib.install_local(**install_args)
         elif isinstance(args, SnapStoreArgument) and args.revision:
-            if which.revision != args.revision:
-                log.info("Ensuring %s snap revision=%s", args.name, args.revision)
-                which.ensure(**install_args)
-                which.hold()
+            log.info("Ensuring %s snap revision=%s", args.name, args.revision)
+            which.ensure(**install_args)
+            which.hold()
```

Comment:
@mateoflorido i think we need to talk about this.  Sorry to call you out to look at it. 

Notice this PR is into the new release_1.32 branch.  I'll open another PR into main.  

but I think `block_refresh` has already checked that the revision isn't changing so it's not necessary to do that here.  I ran into an issue because the `which.revision` was the correct snap revision but the state of the snap was "Available" and not present. 

I can change this to 


```suggestion
            if which.revision != args.revision and which.state != snap_lib.SnapState.Present:
                log.info("Ensuring %s snap revision=%s", args.name, args.revision)
                which.ensure(**install_args)
                which.hold()
```

Reply:
also the snap library `ensure(...)` prevents a refresh if we're on the right revision